fvp-tpm.mk

```diff
@@ -0,0 +1,241 @@
+TF_A_PATH		?= $(ROOT)/trusted-firmware-a
+TF_A_BUILD		?= release
+EDK2_PATH		?= $(ROOT)/edk2
+EDK2_PLATFORMS_PATH	?= $(ROOT)/edk2-platforms
+EDK2_TOOLCHAIN		?= GCC49
+EDK2_ARCH		?= AARCH64
+ifeq ($(DEBUG),1)
+EDK2_BUILD		?= DEBUG
+else
+EDK2_BUILD		?= RELEASE
+endif
+EDK2_BIN		?= $(EDK2_PLATFORMS_PATH)/Build/ArmVExpress-FVP-AArch64/$(EDK2_BUILD)_$(EDK2_TOOLCHAIN)/FV/FVP_$(EDK2_ARCH)_EFI.fd
+FOUNDATION_PATH		?= $(ROOT)/Foundation_Platformpkg
+ifeq ($(wildcard $(FOUNDATION_PATH)),)
+$(error $(FOUNDATION_PATH) does not exist)
+endif
+GRUB_PATH		?= $(ROOT)/grub
+GRUB_CONFIG_PATH	?= $(BUILD_PATH)/fvp/grub
+OUT_PATH		?= $(ROOT)/out
+GRUB_BIN		?= $(OUT_PATH)/bootaa64.efi
+BOOT_IMG		?= $(OUT_PATH)/boot-fat.uefi.img
+FTPM_PATH		?= $(ROOT)/ms-tpm-20-ref/Samples/ARM32-FirmwareTPM/optee_ta
+DEFCONFIG_TSS		?= --br-defconfig build/br-ext/configs/tss
+
+################################################################################
+# Targets
+################################################################################
+all: arm-tf optee-os edk2 grub boot-img
+clean: arm-tf-clean boot-img-clean buildroot-clean edk2-clean grub-clean \
+	ftpm-clean  optee-os-clean linux-ftpm-module-clean
+
+include toolchain.mk
+
+################################################################################
+# Folders
+################################################################################
+$(OUT_PATH):
+	mkdir -p $@
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+TF_A_EXPORTS ?= \
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+TF_A_FLAGS ?= \
+	BL32=$(OPTEE_OS_HEADER_V2_BIN) \
+	BL32_EXTRA1=$(OPTEE_OS_PAGER_V2_BIN) \
+	BL32_EXTRA2=$(OPTEE_OS_PAGEABLE_V2_BIN) \
+	BL33=$(EDK2_BIN) \
+	MBEDTLS_DIR=$(ROOT)/mbedtls  \
+	ARM_ROTPK_LOCATION=devel_rsa \
+	ARM_TSP_RAM_LOCATION=tdram \
+	DEBUG=0 \
+	FVP_USE_GIC_DRIVER=FVP_GICV3 \
+	GENERATE_COT=1 \
+	MEASURED_BOOT=1 \
+	PLAT=fvp \
+	ROT_KEY=plat/arm/board/common/rotpk/arm_rotprivk_rsa.pem \
+	SPD=opteed \
+	TPM_HASH_ALG=sha256 \
+	TRUSTED_BOARD_BOOT=1 \
+	EVENT_LOG_LEVEL=20
+
+arm-tf: optee-os edk2
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) all fip
+
+arm-tf-clean:
+	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) clean
+
+################################################################################
+# fTPM
+################################################################################
+FTPM_FLAGS ?= 						\
+	TA_CPU=cortex-a9				\
+	TA_CROSS_COMPILE=$(AARCH32_CROSS_COMPILE)	\
+	TA_DEV_KIT_DIR=$(OPTEE_OS_PATH)/out/arm/export-ta_arm32 \
+	CFG_TA_DEBUG=y CFG_TEE_TA_LOG_LEVEL=4 CFG_TA_MEASURED_BOOT=y
+.PHONY: ftpm
+ftpm: optee-os
+	$(FTPM_FLAGS) $(MAKE) -C $(FTPM_PATH)
+	-mkdir -p $(BR2_ROOTFS_OVERLAY)/lib/optee_armtz/
+	@cp -f $(FTPM_PATH)/out/fTPM/bc50d971-d4c9-42c4-82cb-343fb7f37896.ta \
+		$(BR2_ROOTFS_OVERLAY)/lib/optee_armtz/
```

Comment:
I don't think it is nice to install something in build source tree. The TA should preferably be installed straight in Buildroot output target/lib/optee_armtz/, preferably through a BR package, i.e. creating necessary files in build/br-ext/package/ftpm/.

Reply:
Hi @etienne-lms ,

I added to new packages for Buildroot:

- ftpm_optee_ext to install the ftpm TA and
- linux_ftpm_mod to install the TPM Kernel module.

The packages don't build the binaries. They just install them once they are built (similarly to what optee_os_ext does).

Cheers,
Javier